app/src/pages/settings/CreateSystemAPIKeyDialog.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useMutation } from "react-relay";
+import { Form } from "react-router-dom";
+import { graphql } from "relay-runtime";
+
+import {
+  Alert,
+  Button,
+  Dialog,
+  Flex,
+  TextArea,
+  TextField,
+  View,
+} from "@arizeai/components";
+
+import { CreateSystemAPIKeyDialogMutation } from "./__generated__/CreateSystemAPIKeyDialogMutation.graphql";
+
+export type SystemKeyFormParams = {
+  name: string;
+  description: string | null;
+};
+
+/**
+ * A dialog that allows admin users to create a system API key.
+ * TODO: Add expiry date field
+ */
+export function CreateSystemAPIKeyDialog(props: {
+  onSystemKeyCreated: (jwt: string) => void;
+}) {
+  const { onSystemKeyCreated } = props;
+  const [error, setError] = useState<string | null>(null);
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid },
+  } = useForm<SystemKeyFormParams>({
+    defaultValues: {
+      name: "System",
+      description: "",
+    },
+  });
+
+  const [commit, isCommitting] = useMutation<CreateSystemAPIKeyDialogMutation>(
+    graphql`
+      mutation CreateSystemAPIKeyDialogMutation(
+        $name: String!
+        $description: String = null
+      ) {
+        createSystemApiKey(input: { name: $name, description: $description }) {
+          jwt
+          query {
+            ...SystemAPIKeysTableFragment
+          }
+        }
+      }
+    `
+  );
+
+  const onSubmit = useCallback(
+    (data: SystemKeyFormParams) => {
+      setError(null);
+      commit({
+        variables: data,
+        onCompleted: (response) => {
+          onSystemKeyCreated(response.createSystemApiKey.jwt);
+        },
+        onError: (error) => {
+          setError(error.message);
+        },
+      });
+    },
+    [commit, onSystemKeyCreated]
+  );
+
+  return (
+    <Dialog title="Create a System Key" isDismissable>
+      {error && (
+        <Alert variant="danger" banner>
+          {error}
+        </Alert>
+      )}
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <View padding="size-200">
+          <Controller
+            name="name"
+            control={control}
+            rules={{
+              required: "System key name is required",
+            }}
```

Comment:
for my product level understanding - there can be multiple system keys, and they can all have different names, correct?

Reply:
Yup, and they can ultimately have the same name if they want